[Yu Xuan] iP

yuxuanleong opened the pull request:
# DukePlus
An **all in one** app for you to keep track of your tasks 👍 

Types of tasks that can be added
* todo
* event
* deadline

> Tips: date and time can be added to event tasks

You can try it out by,
1. Downloading it from [here](https://github.com/yuxuanleong/ip)
2. Click on the latest release
3. Click on the *duke.jar* file 

Duke versions:
- [X] Duke
- [ ] DukePlus
- [ ] DukePro 

You can run it in your IDE. Here's the `main` method
```java
public class Duke {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

yuxuanleong committed:
modified Duke.java (Level-1)

yuxuanleong committed:
modified Duke.java (Level-2)

yuxuanleong committed:
Added Task.java, modified Duke.java (Level-3)

yuxuanleong committed:
Added Deadline/Event/ToDos. Modified Duke (Level -4)

yuxuanleong committed:
Exception handling (Level -5)

yuxuanleong committed:
Delete function

yuxuanleong committed:
TextUITesting

yuxuanleong committed:
Level - 7 (Save)

yuxuanleong committed:
Improvements

yuxuanleong committed:
Format date and time

yuxuanleong committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Improvements

yuxuanleong committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Format date and time

yuxuanleong committed:
Fix bugs

yuxuanleong committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into add-gradle-support

yuxuanleong committed:
Merge branch 'add-gradle-support'

yuxuanleong committed:
More OOP

yuxuanleong committed:
Packages

yuxuanleong committed:
JUnit test

yuxuanleong committed:
Level-9

yuxuanleong committed:
Merge branch 'branch-Level-9'

yuxuanleong committed:
JavaDocs

yuxuanleong committed:
Adds JavaDocs

yuxuanleong committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

yuxuanleong committed:
Fixed some errors

yuxuanleong committed:
Coding standard

yuxuanleong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

yuxuanleong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

guowei42 commented on src/test/java/duke/task/ToDosTest.java:
```diff
@@ -0,0 +1,18 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Is is possible to do without this whole package import?

guowei42 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,148 @@
+public class Ui {
+
+    private static String INDENT = "     ";
+    private static String LINE = "    _______________________________________________________";
```
Since this is a constant, best to add a 'final here as well'.

guowei42 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,148 @@
+    private static String LINE = "    _______________________________________________________";
+
+    public Ui() {
+    }
```
I believe this can be removed?

guowei42 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+        try {
+            FileWriter filewriter = new FileWriter(dataFile.getPath());
+
+            for (Task task : taskList.getList()) {
```
I feel a bit uneasy with this `getList()` public method since if anyone can access this, they can freely modify it as well. Alternatively, I was thinking this can be replaced with getting the total length of the taskList, then getting the tasks separately from the index with your `getTask()` method. This way, the getList() method is not needed anymore since I think its only used once here.

guowei42 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,76 @@
+        return "[E]"
+                + super.toString()
+                + " (at: "
+                + LocalDate.parse(date).format(DateTimeFormatter.ofPattern("dd MMM yyyy"))
```
What will happen if LocalDate is not able to parse the date input by the user?

guowei42 reviewed (commented):
I really like the structure of the classes and method, just noted some queries and alternatives to consider.

RezwanArefin01 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Represents Exceptions that are specific to Duke program.
+ */
+public class DukeException extends Exception{
```
'{' should be preceded by a whitespace.

RezwanArefin01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,64 @@
+
+    private boolean isRunning = true;
+
+    public Duke(String filePath) {
```
JavaDoc missing. I have noticed the same in several other places too, for public constructors.

RezwanArefin01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        }
+    }
+
+}
```
Each file should end with a newline character. I have noticed this in some other files too.

RezwanArefin01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        String userInput = input.trim();
+        if (userInput.equals("list")) {
+            ui.printTaskList(taskList);
+        }else if (containsOperationWord(userInput)) {
```
'}' should be followed by a whitespace.

RezwanArefin01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+            String firstWord = tokens[0];
+
+            switch (firstWord) {
+            case "find" :
```
Extra whitespace before ':'.

RezwanArefin01 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,43 @@
+    private String name;
+
+    public Task(String name) {
+       this.done = false;
```
Indentation should be 4 spaces.

RezwanArefin01 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+            if (i == list.size() - 1) {
+                text += i + 1 + ". " + list.get(i);
+            } else {
+                text += i + 1 + ". " + list.get(i) +"\n     ";
```
'+' should be followed by a whitespace.

RezwanArefin01 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+            if (i == resultList.size() - 1) {
+                text += i + 1 + ". " + resultList.get(i);
+            } else {
+                text += i + 1 + ". " + resultList.get(i) +"\n     ";
```
'+' should be followed by a whitespace.

RezwanArefin01 reviewed (commented):
Looks good. :+1:  
Some changes regarding coding standard suggested.

yuxuanleong committed:
Improves on documentation

yuxuanleong committed:
GUI

yuxuanleong committed:
GUI part 2

yuxuanleong committed:
Merge branch 'branch-Level-10'

yuxuanleong committed:
Improves on GUI

yuxuanleong committed:
Add assert statements

There are no assert statements.

Assertions are needed to check whether assumptions have been violated.

Add assert statements to getTask, deleteTask, save methods.

Improve code quality at the same time.

yuxuanleong committed:
Improves code quality

Long methods and arrow-head mehtods.

Make the code more readable and simple to understand.

Nested methods are being extracted out.

yuxuanleong committed:
Merge pull request #2 from yuxuanleong/branch-A-Assertions

Add assert statements

yuxuanleong committed:
Merge pull request #3 from yuxuanleong/branch-A-CodeQuality

Improves code quality

yuxuanleong committed:
Improves on date and time formatting

yuxuanleong committed:
Adds an extension to sort the tasks

yuxuanleong committed:
Merge branch 'branch-BCD-Extension'

yuxuanleong committed:
Deletes datetime comparator files

yuxuanleong committed:
Adds UG

yuxuanleong committed:
Adds Ui.png

yuxuanleong committed:
Creates JAR file

yuxuanleong committed:
Adds an instruction message when program is initialised

yuxuanleong committed:
Update JavaDocs

yuxuanleong committed:
Improves GUI

yuxuanleong committed:
Update README.md

yuxuanleong committed:
Fix JavaDocs

yuxuanleong committed:
Upgrade GUI

make it resizable, add app icon and title, minor changes to help message

yuxuanleong committed:
Add clear command

yuxuanleong committed:
Update Ui.png

yuxuanleong committed:
Add clear command documentation

yuxuanleong committed:
Resize Ui.png

yuxuanleong committed:
Update README.md

yuxuanleong committed:
Update README.md
